### ISSUE TITLE
English support

### ISSUE THREAD

**Issue by mawek** [siyuan-note/siyuan#7758]
### 在什么情况下你需要该特性？In what scenarios do you need this feature?

I'm logseq user and I'm looking for alternatives. 
This project seems interesting feature-wise, but what bothers me the most is lack of english resources. 
Though main page and readmes are in english, all other resources are not - issues, communities, discussion.. 
Therefore, as playground it is very nic, but it's hard to commit to it since long term support is kind of hard requirements in such tools.

### 描述可能的最优解决方案 Describe the optimal solution

The best solution solution would be switching to english completely - not sure how viable this is :)

**Comment by nekrondev** [comment]
@mawek From my personal experience both devs can be contacted via English language w/out any problems (they both speak it very well and it's not like English for runaways). 

SiYuan's Discourse server offers an English speakers room, issues submitted here at Github can also be written down in English.
I think SiYuan has many Chinese native speakers so it's a bit challenging for us non-Chinese speakers, but Yandex translation or [deepl.com](https://deepl.com) works great for translation between both languages. There is also a [Reddit](https://www.reddit.com/r/SiyuanPKM/) `/r/SiYuanPKM` where you can post questions in English language.

Nonetheless my **absolute pro-tip** is: Just start using SiYuan! You'll soon discover new features (you can call it batteries included as it offers so much features out of the box), setup S3 backup and synchronization to different devices (if you use e.g. desktop application and a self-hosted docker-based instance) and you're ready to go. SiYuan syncs your notes in the background and you can restore snapshots of your notes any time in case something breaks.

Enjoy using this great tool. It's a game changer, trust me.

**Comment by mawek** [comment]
I don't dispute anything you said. 

Nevertheless, it's still hard to commit to it. Community and visibility is important for such tools. 

For example - in logseq (or any other similar tool) I often like wonder around to see what's going on, to check roadmap, check what issues it solved lately, how is development proceedeing, whether it aims for stability and bugfixes, or just adding new features,.. I like to read forums and see how other users interact with it. I need to gain confidence that it's not just some one-man show that might stop working in one year (I'm not saying this is the case)

I can use translator here and there (to translate some selection via deepl chrome extension for example) but it doesn't work seamlessly.

I personally had used few note taking apps in the past, always migrating all data to something new "better", so now I'm cautious and i think twice before i migrate again :)

I don't expect that you rewrite everything to english or anything. I just wanted to share my 2 cents on this.

**Comment by 88250** [comment]
Thanks for sharing your views, as one of the developers, let me briefly describe our situation.

Another core developer @Vanessa219 is my wife, we are currently living in Yunnan Province, China. In the almost 3 years since SiYuan was created, we have spent a lot of time on architecture design and code implementation, so we have no time to think about internationalization. Everything is done as quickly as possible, including code comments and change logs, community discuss and so on.

To be honest, judging from the current degree of completion of SiYuan, we did not expect to attract the attention of foreign users, because there are many projects that are much better than SiYuan on the international stage, and SiYuan still has a long way to go. So we still have to concentrate on technology at present, and the community relies on users to build spontaneously.

Maybe in the near future, we will establish a separate English discussion forum similar to ld246.com, and provide subscription services in non-mainland China. Before that, you are welcome to communicate in English on GitHub Issues. We also hope that SiYuan users (including Chinese users) can use English on GitHub Issues, which is conducive to the development and growth of SiYuan, thank you.

**Comment by Cyber-Syntax** [comment]
@88250 "because there are many projects that are much better than SiYuan"
Do not underestimate your project, because it is the only alternative to the current notion. Yes, there are a lot of new features that need to be added to get the proper notion alternative, but there are currently no alternatives because they are in development (e.g appflowy, affine). I think I suggest you reconsider going international because it will increase competition with other alternatives and maybe provide some motivation.

**Comment by nhan000** [comment]
I'm chiming in here, hoping you'd reconsider the internationalization aspect.

- When you go international, you can leverage plugin developers to build features that enhance your app without having to build it yourself. Without its plugins, many wouldn't use Obsidian. If you don't start soon, I'm afraid people would just concentrate towards other apps. People would also create tutorials and guides without you having to do everything.
- I think you're in a good place to compete against Logseq, which has slowed down its development significantly while you're still going very strong. Many *many* people are looking for alternatives, or already switched away from Logseq, including me, because the app is so slow it's unusable (and the countless annoying bugs happened with nearly every update). You have PDF annotations and it's free! That alone was the reason I switched to Logseq in the first place. There're so many things you're doing right: proper management of GitHub issues, apps on major app stores, etc. And I think it's a pity that you can't capture more of the market because of the language barrier.

At the end, I understand the constraints of time by a small team and your astute focus on carving out a niche in the Chinese-speaking community (I heard that Logseq's Chinese community is very anemic. The quality of its English community is also not great). But I do think there's a gap in the market for your app, so I hope you would consider doing a few things that would at least not scaring away non-Chinese users. I didn't even know the main team has 2 devs and speak English fluently until today. Obsidian also started with just 2 people and they are quite involved with the community, but gradually they were able to build a strong volunteer community that helped them a lot.

**Comment by alanjustalan** [comment]
> I'm chiming in here, hoping you would reconsider the internationalization aspect.
> 
> * When you go international, you can leverage plugin developers to build features that enhance your app without having to build it yourself. Without its plugins, many wouldn't use Obsidian. If you don't start soon, I'm afraid people would just concentrate towards other apps. People would also create tutorials and guides without you having to do everything.
> * I think you're in a good place to compete against Logseq, which has slowed down its development significantly while you're still going very strong. Many _many_ people are looking for alternatives because the app is so slow it's unusable and the countless annoying bugs happened with nearly every update. You have PDF annotations and it's free! That alone was the reason I switched to Logseq in the first place. I think it's a pity that you can't capture more of the market because of the language barrier. There're so many things you're doing right:
>   
>   * Proper management of GitHub issues
>   * Presence on major app stores
>   * You can proudly say that "All local features are free." Logseq [can't even say that](https://discuss.logseq.com/t/what-is-logseqs-business-model/389/78) which has turned away many good contributors. Recently with the demise of other open-source notetaking projects such as Athens and Dendron, I am worried that Logseq would come to an end too.
>   * Support multi-tab: Way more user friendly than Logseg. Logseq always has the feeling of being finicky and you have to work around its limitation.
>   * Sync has been figured out while Logseq has wrangled it for a year and it still deletes the users' notes.
>   * And so many other things I won't bother you with.
> 
> At the end, I understand the constraints of time by a small team and your astute focus on carving out a niche in the Chinese-speaking community. I heard that Logseq's Chinese community is too anemic that some have to flee to the English community and use a Translator to ask for help (The quality of its English community is also not great though. Logseq devs are no longer responsive to users' feedback as before). Obsidian also started with just 2 people and they have been quite involved with the community, but gradually they were able to build a strong volunteer community that helped them a lot.
> 
> I do think there's a gap in the market for your app, so I hope you would consider doing a few things that would at least not scaring away non-Chinese users and developers. I didn't even know the main team has 2 devs and speak English fluently until today. Maybe starting with curating a few Chinese tutorials that clearly showcase the special features of the app and translating them into English?

I agree with your point of view that the development of widgets, plugins and various usage methods for Simplenote requires 
several times more users than China. Moreover, it is difficult to migrate double-linked notes once they are fixed and hardly changeable. Therefore, it is better to seize users as early as possible when the product approaches maturity. I hope that D大 can reconsider the issue of internationalization development.    @88250

**Comment by InEase** [comment]
I agree with the conversations above. SiYuan with its great quality absolutely worth a larger community including users other than Chinese. And we definitely need to build a community in English for non-Chinese speakers. Migrating between double-linked notes are diffcult, so maybe we should begin now and I'm glad to help. I list some TODOs here and maybe we can start here.

* Setup a Community for non-Chinese speakers

  * [X] we have built a [[Discord Server](https://discord.com/invite/bzfCBwMzdP)](https://discord.com/invite/bzfCBwMzdP), [[Twitter Account](https://twitter.com/SiYuanNotes)](https://twitter.com/SiYuanNotes), but it seems they are lack of updates
  * [ ] Maybe we need English speakers as moderators and content creators to support our English community.
* Create a variety of English learning resources, such as videos, articles, etc.

  * [ ] this is pretty difficult at the beginning i think, maybe we can start with translating some Chinese materials?
* Encourage user-generated content and peer feedback in English, such as blogs, comments, reviews, etc.
* Monitor and analyze user behavior and feedback to improve the user experience and retention of the English community.

**Comment by nhan000** [comment]
Hi @88250, I can help with managing/monitoring the Discord community and the Twitter account (I won't do much though, but I will keep it alive). I have DM'ed you on Discord and another 2 moderators. Let's find a place somewhere (perferably the mod channel in the current SiYuan's Discord server?) to make a plan about this.

**Comment by 88250** [comment]
@nhan000 First of all, thank you for your enthusiasm, but I have to explain a few things:

* The Discord and Twitter mentioned above were not established by us, so I actually don’t have account permissions, you can try to communicate with the administrator
* It may be too early to establish an official English community. We plan to internationalize the GitHub repository first. The main action steps include:
   * Convert existing Issues translations to English
   * New Issues are also in English
   * The code commit message is in English
   * Release changelog in English
  
After these steps have been implemented for a period of time, we will see if we can attract more international friends to help us improve this project, and then we will formally establish an official English discussion forum.

The more optimistic expectation is that after the internationalization of the project is successful, we will further consider open source commercialization, that is, provide official cloud services in non-mainland China regions and start selling subscriptions. The current general plan is as follows:

* Open to a limited number of lifetime subscriptions
* Open early bird price subscriptions
* Open Regular price subscriptions

In short, the most important thing is internationalizing the GitHub repo first. After all, the technical implementation is the foundation of everything, thank you.

**Comment by 88250** [comment]
I closed this issue, but we are still welcome to continue discussing the details of SiYuan's internationalization, thank you.

**Comment by InEase** [comment]
Appreciate D and V again for their great work! I agree with D's comment above, we can first internationalize the GitHub repository as it is easier and can provide useful information for the non-Chinese community. I have noticed that some issues' titles have been translated to English by D, which is really impressive.

I have a small suggestion when translating the repo - maybe we should keep both Chinese and English titles when internationalizing? The Chinese titles are also important for users to search for information.

Meanwhile, I wonder if there exists some ways for us, the community, to help internationalize the GitHub repo as it is a lot of work.

**Comment by nhan000** [comment]
I'd chime in here as well:
-  The Chinese users are your base, please don't turn against them. As a user it feels terrible when the devs seem to stop caring about your needs. They are your loyal users, and they feel betrayed. Again, you catering to Chinese-speaking users is a smart decision. I started looking for other options the moment I felt my valid feedback wasn't heard by Logseq devs anymore. Please, please reconsider this. 
    - Depend on the level of automation, at the minimum, make the title bilingual, and that is for now already good enough. That would make searching easy for the English speaking people. After that they can use the translator built into every modern browser to understand what that issue is about. Make them adapt to the Chinese root of your app. Not the other way around.
- I'd also pin this English support to the top of the GitHub issue for reference by any English users coming across the issue page. (Or at least leave a link in the Readme page so that they can follow your internationalization effort.)

**Comment by HowcanoeWang** [comment]
Maybe we can use the 'github discussion' as the English community forum and keep 链滴 as Chinese community forum?

**Comment by JARVISia** [comment]
Perhaps we should keep this issue open for better discussion?

**Comment by 88250** [comment]
@nhan000 I agree with you, SiYuan Chinese users currently account for the vast majority, we cannot prevent the use of Chinese Feedback Issues for the sake of internationalization.

For this reason, I think a more compromise solution is to establish a separate Chinese repo for issues. If there is a Chinese issue feedback on the main repo, we will move the Chinese issue to the Chinese repo, which can not only keep track, but also bring Better internationalization/localization experience.

**Comment by WolkeAtreide** [comment]
> @nhan000首先，感谢您的热情，但我必须解释几件事：
> 
> * 上面提到的Discord和Twitter不是我们建立的，所以我实际上没有帐户权限，您可以尝试与管理员沟通
> * 现在建立一个官方的英语社区可能还为时过早。我们计划首先将 GitHub 存储库国际化。主要操作步骤包括：
>   * 将现有问题翻译转换为英语
>   * 新刊也有英文版
>   * 代码提交消息为英文
>   * 英文版发布更新日志
> 
> 这些步骤实施一段时间后，我们会看看能不能吸引更多的国际朋友来帮助我们完善这个项目，然后我们会正式建立一个官方的英语讨论论坛。
> 
> 比较乐观的期待是，在项目国际化成功后，我们会进一步考虑开源商业化，即在非中国大陆地区提供官方云服务，开始销售订阅。目前的总体规划如下：
> 
> * 开放有限数量的终身订阅
> * 开放早鸟价订阅
> * 开放正常价格订阅
> 
> 简而言之，最重要的是首先将 GitHub 存储库国际化。毕竟，技术实现是一切的基础，谢谢。

国内的黄牛党一下子兴奋了

**Comment by 88250** [comment]
We're closing this issue first, and will keep it pinned on Issues for a while.

**Comment by nhan000** [comment]
Just a quick update 

- Official English forum is at https://liuyun.io/
- SiYuan [2.9.3](https://github.com/siyuan-note/siyuan/releases/tag/v2.9.3) supports sync outside of mainland China and is offering a limited number of lifetime subscriptions.
- SiYuan 2.9.3 also supports importing [[WikiLinks]] from Obsidian, Logseq, etc.

**Comment by sagarbehere** [comment]
@88250 , I just discovered SiYuan and am very excited about it. I joined the Official English Forum at https://liuyun.io/ but am unable to post my questions because it seems like I need “20 ask rewards points” in order to post?!? So I am asking this question here: What are “ask points” and how can I get them? In general, I think it is a pleasant and positive user experience if they can create an account on the English forum and start asking questions immediately, without having to get any “ask points”. So please consider this.

**Comment by 88250** [comment]
@sagarbehere Hello, please send us your forum username so that we can diagnose the problem. If it is not convenient to make it public, you can send it to the email address 845765@qq.com. Thank you.
